plugins/source-verifier/api.ts

```diff
@@ -1,16 +1,34 @@
-export interface Network {
-  name: 'ropsten' | 'mainnet' | 'rinkeby' | 'goerli'
-  id: number  
-}
-
 export interface VerifiedArtifacts {
   metadata: any
   contracts: any
 }
 
+export interface VerifyResponse {
+  match: 'partial' | 'perfect' | 'no match',
+  success: boolean,
+  message: string
+}
+
+export interface VerifyData {
```

Comment:
@GrandSchtroumpf @yann300 in the end this is all the stuff that is needed.
Network can be obtained from remix API so I can create another verifyWithRemixNetwork (better name needed) but uniform verify should have all this parameters.

Reply:
if you want to have an API that require a network as second parameter, the name can be `verifyByNetwork` and `verify` would just use the network from the context (same apply with `fetch`)